nomad/structs/structs.go

```diff
@@ -9935,6 +9935,24 @@ func (a *Allocation) DisconnectTimeout(now time.Time) time.Time {
 	return now.Add(*timeout)
 }
 
+// SupportsDisconnectedClients determines whether both the server and the task group
+// are configured to allow the allocation to reconnect after network connectivity
+// has been lost and then restored.
+func (a *Allocation) SupportsDisconnectedClients(serverSupportsDisconnectedClients bool) bool {
+	if !serverSupportsDisconnectedClients {
```

Comment:
Not particularly related to this PR but it occurs to me that we should have a deprecation plan for tearing out this version check that's all over the place. That is, what version of Nomad can we safely remove this check? We have a very sad check for nodes > v0.8 for client RPCs (and a few other checks floating around) that we've been living with for a while and should probably remove, but it'd nice to have plan to remove on minimum version checks going forward.

Reply:
👍